feat: improved experience on DNSLink websites

lidel opened the pull request:
This PR changes default behavior and disables redirect of DNSLink websites
to local gateway, as noted in https://github.com/ipfs-shipyard/ipfs-companion/issues/701#issuecomment-562304730

## Context and motivation

- In the past, redirect of DNSLink websites to local gateway was enabled by default.
  - User had to opt-out via redirect opt-out per hostname.
- #667 discussed problematic situation of breaking Origin of DNSLink websites.
- #701 describes agreed temporary solution 
  - temporary until something like subdomain gateway on `*.localhost` (https://github.com/ipfs/go-ipfs/issues/6498) or Web Package (https://github.com/ipfs/in-web-browsers/issues/121) lets us redirect without breaking Origin isolation.

### TODO
- [x] disable DNSLink redirect by default
- [x] display IPFS actions on DNSLink websites (without redirect)
- [x] add DNSLink section to Preferences screen
  - [x] enabling DNSLink redirect is off by default, requires explicit opt-in. 
  - [x] If user enables it, we warn about breaking of Origin sandbox:
    > ![2019-12-05--23-22-39](https://user-images.githubusercontent.com/157609/70279197-3b9f3e00-17b6-11ea-9dad-90f2d60b4a70.png)
- [ ] ensure IPFS recovery works if DNSLink's HTTP server is down
- [x] update browser action menu to display item for opening DNSLink website at Custom Gateway (in a new tab) (https://github.com/ipfs-shipyard/ipfs-companion/pull/826/commits/963a3406687afc17046cfb6f0192ba1765a4431b)
   - [x] remove per-domain redirect toggle from main menu (https://github.com/ipfs-shipyard/ipfs-companion/pull/826/commits/c7c32216a98e80cace6b7aae5498658d14b0edb7)

Closes #701, #667, helps with https://github.com/ipfs/docs/issues/271
cc @cwaring @hugomrdias @autonome

cwaring commented on add-on/_locales/en/messages.json:
```diff
@@ -275,6 +275,18 @@
     "description": "An option description on the Preferences screen (option_useCustomGateway_description)"
   },
+  "option_dnslinkRedirect_title": {
+    "message": "Force load from Custom Gateway",
```
I agree with the language tweak here. This might be out of scope for this PR but do you think it's widely understood what a Custom Gateway is? I would be inclined to be more specific, such as: **Always load DNSLink websites via IPFS Companion's local gateway.** (if that is a correct statement?)

lidel replied:
Will tweak the label, but want to keep "force" for now to emphasize its "over the top" behavior (until we can redirect in a way that keeps origin sandbox) 

@cwaring ack, lets track that as part of https://github.com/ipfs-shipyard/ipfs-companion/issues/741

lidel commented:
Simplified main menu and added option to `View on Gateway` which opens current tab via Gateway in a new tab:

> ![view-on-gateway2019-12-06--20-02-43](https://user-images.githubusercontent.com/157609/70349328-d191a200-1864-11ea-999b-867b1be3dbee.png)

Due to https://github.com/ipfs-shipyard/ipfs-companion/issues/808  my plan is to fast-track this to Beta (and then Stable in a day or two), but feel free to provide feedback here or in new issue, we can improve UI in new PR.

momack2 commented:
Needs review and explicit approval by @jbenet

lidel commented:
The second leg of this change is in https://github.com/ipfs-shipyard/ipfs-companion/pull/827, where preload of visited DNSLink URLs to local node is added to ensure content is still distributed and is available if original HTTP server is down.
